src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -3153,7 +3153,14 @@ private static bool IsReallyAType(TypeSymbol? type)
 
         private static void GetAllCandidates(Dictionary<TypeWithAnnotations, TypeWithAnnotations> candidates, ArrayBuilder<TypeWithAnnotations> builder)
         {
-            builder.AddRange(candidates.Values);
+            // Not using builder.AddRange here because the dictionary values enumerator is a struct, and calling AddRange will have to box the enumerator.
+            // Instead, we increase the builder capacity, then loop over the values.
+            // Also, we don't access candidates.Values to avoid realizing the Dictionary's internal ValueCollection (it's created lazily when Values is accessed)
+            builder.Count += candidates.Count;
```

Comment:
>builder.Count += candidates.Count; [](http://example.com/codeflow?start=12&length=34)

This adds a bunch of empty items to the builder. I assume the intent is to increase capacity. Consider using ```EnsureCapacity``` instead.